std/process.d

```diff
@@ -1434,28 +1429,39 @@ version (Windows) @system unittest
 // (checking that it is in fact executable).
 version (Posix)
 private string searchPathFor(scope const(char)[] executable)
-    @trusted //TODO: @safe nothrow
+    @safe
 {
     import std.algorithm.iteration : splitter;
-    import std.conv : to;
-    import std.path : buildPath;
+    import std.conv : text;
+    import std.path : chainPath;
 
-    auto pathz = core.stdc.stdlib.getenv("PATH");
-    if (pathz == null)  return null;
+    string result;
 
-    foreach (dir; splitter(to!string(pathz), ':'))
-    {
-        auto execPath = buildPath(dir, executable);
-        if (isExecutable(execPath))  return execPath;
-    }
+    environment.getImpl("PATH",
+        (scope const(char)[] path)
```

Comment:
The callback here is not `@safe` and apparently fails to compile with ldc-1.24.0 on Linux-32.
```
cannot pass argument `__lambda2` of type `void delegate(scope const(char)[] path) @system` to parameter `scope void delegate(const(char)[]) @safe sink`
```

Reply:
After making it `@safe`
`std/process.d(1448): Error: scope variable dir assigned to execPath with longer lifetime`

Reply:
If I add `@safe` explicitly, it compiles for me.

How can I reproduce this error?

What error do you get if you add an explicit `@safe` annotation?

Reply:
> How can I reproduce this error?

Reproduced with `digger run master -- dmd -o- -preview=dip1000 process.d`

Reply:
Looks like something changed in `dip1000` on master. Not yet sure if this is a regression. Bisecting.

Reply:
BTW, feel free to revert this if it's blocking anything.

Reply:
Wow, it was actually caused by a regression in Phobos.

Introduced in #7719.

I suggest reverting that pull request, since it was intended as a regression.

Reply:
https://github.com/dlang/phobos/pull/7730

Reply:
See https://github.com/dlang/phobos/pull/7731 @CyberShadow

Reply:
I think we should prefer #7730 as that addresses a regression visible from outside Phobos.